tools/golden_tests_harvester/lib/golden_tests_harvester.dart

```diff
@@ -2,42 +2,98 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-import 'dart:io';
+import 'dart:io' as io;
 
 import 'package:path/path.dart' as p;
 import 'package:skia_gold_client/skia_gold_client.dart';
 
-import 'logger.dart';
-
-export 'logger.dart';
-
-/// Reads the digest inside of [workDirectory], sending tests to
-/// [skiaGoldClient].
-Future<void> harvest(SkiaGoldClient skiaGoldClient, Directory workDirectory,
-    List<Object?> entries) async {
-  await skiaGoldClient.auth();
+import 'src/digests_json_format.dart';
 
+/// Uploads the images of digests  in [workDirectory] to Skia Gold.
+///
+/// The directory is expected to match the following structure:
+/// ```txt
+/// workDirectory/
+///   - digests.json
+///   - test_name_1.png
+///   - test_name_2.png
+///   - ...
+/// ```
+///
+/// The format of `digests.json` is expected to match the following:
+/// ```jsonc
+/// {
+///   "dimensions": {
+///     // Key-value pairs of dimensions to provide to Skia Gold.
+///     // For example:
+///     "platform": "linux",
+///   },
+///   "entries": [
+///     // Each entry is a test-run with the following format:
+///     {
+///       // Path must be a direct sibling of digests.json.
+///       "filename": "test_name_1.png",
+///
+///       // Called `screenshotSize` in Skia Gold (width * height).
+///       "width": 100,
+///       "height": 100,
+///
+///       // Called `differentPixelsRate` in Skia Gold.
+///       "maxDiffPixelsPercent": 0.01,
+///
+///       // Called `pixelColorDelta` in Skia Gold.
+///       "maxColorDelta": 0
+///     }
+///   ]
+/// }
+/// ```
+///
+/// If [dryRun] is true, the images will _not_ be uploaded to Skia Gold, and
```

Comment:
`dryRun` is not apropos to `harvest`, I think you refactored this and didn't yet update the docstring.

Reply:
Right, thanks for catching the refactor!